### PR TITLE
Counts by Pharmacy First Service and Clinical Condition

### PR DESCRIPTION
**Pull request by viv3ckj**

### CODE REVIEW

File: analysis/reports/pf_report.Rmd [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+```
+
+```{r, message=FALSE, warning=FALSE}
+plot_services <- plot_measures(df_measures,
+title = "Number of consultations for each clinical service per month",
+measure_names = names(clinical_service_labels),
+custom_labels = clinical_service_labels,
+y_label = "Number of codes for consultations",
+)
+print(plot_services)
```
Comment (milanwiedemann) [inline analysis/reports/pf_report.Rmd:125]: see comment below
```suggestion
plot_measures(
  df_measures,
  title = "Number of consultations for each clinical service per month",
  measure_names = names(clinical_service_labels),
  custom_labels = clinical_service_labels,
  y_label = "Number of codes for consultations",
)
```

File: analysis/reports/pf_report.Rmd [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+
+## Background <a name="background"></a>
+
+Add background here.
+
+<br>
+
+## Methods <a name="methods"></a>
+
+This study used data from OpenSAFELY-TPP, which covers 40% of the population of England. For a description of the representativeness of this sample, please see our manuscript [here](https://wellcomeopenresearch.org/articles/7-191/v1). Individuals were included if they were alive and registered at a TPP practice each month, across the study period. Patients were excluded if their listed age was not between 0 and 120 years.
```
Comment (milanwiedemann) [inline analysis/reports/pf_report.Rmd:46]: you could have a look at how citations work in R Mardown and see if you can add this paper as a citation? have a look at this if this https://bookdown.org/yihui/rmarkdown-cookbook/bibliography.html if you have time to add, but not essential at this stage

File: lib/functions/function_plot_measures.R [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    axis_x_text_size = 7) {
+
+  # Check if the necessary columns exist in the data
+  if (date_col %in% names(data) == FALSE) {
+    stop("Data does not have a column with the name '", date_col, "'")
+  } else if (value_col %in% names(data) == FALSE) {
+    stop("Data does not have a column with the name '", value_col, "'")
+  } else if (measure_col %in% names(data) == FALSE) {
+    stop("Data does not have a column with the name '", measure_col, "'")
+  }
```
Comment (milanwiedemann) [inline lib/functions/function_plot_measures.R:39]: fantastic work!!!!

File: lib/functions/function_plot_measures.R [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+  # Create plot
+  plot1 <- ggplot(
+    data,
+    aes(
+      x = !!date_sym,
+      y = !!value_sym,
+      color = !!measure_sym,
+      group = !!measure_sym
+    )
```
Comment (milanwiedemann) [inline lib/functions/function_plot_measures.R:68]: this is also great, but I think can follow more recent tidyeval implementation using `{{ }}` but I have to double check, for now this is perfect but I might change it later!

File: analysis/reports/pf_report.Rmd [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+  * [Urinary tract infection etc.](#urt)
+* [Hypertension](#hyp)
+  * [Age](#hyp_age)
+  * [Sex](#hyp_sex)
+  * [Region](#hyp_region)
+* [Contraception](#contra)
+  * [Age](#contra_age)
+  * [Region](#contra_region)
+
+<br>
```
Comment (milanwiedemann) [inline analysis/reports/pf_report.Rmd:83]: this is now automatically created using the `toc` flag in the yaml?

```suggestion
```

File: analysis/report_measures.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
+from ehrql import INTERVAL, create_measures, months, codelist_from_csv
+from ehrql.tables.tpp import clinical_events, patients, practice_registrations
```
Comment (milanwiedemann) [inline analysis/report_measures.py:2]: `patients` table is not used in this definition, but we might want to use variables like age or sex from this later?

File: analysis/reports/pf_report.Rmd
Comment (milanwiedemann) [inline analysis/reports/pf_report.Rmd:1]: I'd add this as a top level directory of the project, not a subdir of the analysis folder. also, we should add the output of this notebook to the `.gitignore` file as it's not needed in the GH repo

File: analysis/reports/pf_report.Rmd (hunk 5 above)
Comment (viv3ckj) [reply analysis/reports/pf_report.Rmd:83]: Yup I need to remove it!

File: analysis/report_measures.py (hunk 6 above)
Comment (viv3ckj) [reply analysis/report_measures.py:2]: Yep, we should be using it later for sex and age.

File: lib/functions/function_plot_measures.R (hunk 4 above)
Comment (viv3ckj) [reply lib/functions/function_plot_measures.R:68]: I wasn't able to get this to work, so maybe you can look into it?

**Comment by viv3ckj** [comment]
closes #5 and #2

**Pull request merged**